### ISSUE TITLE
Fix new TeamCity violations

### ISSUE THREAD

**Issue by rnveach** [checkstyle/checkstyle#3616]
https://teamcity.jetbrains.com/viewLog.html?buildId=940313&tab=Inspection&buildTypeId=Checkstyle_IdeaInspectionsPullRequest

>  Number of inspection errors 1255 is 1255 more than the provided threshold 0
Number of inspection warnings 1126 is 1126 more than the provided threshold 0 

I believe this is because of the new release of TC.
These violations must be fixed, or suppressed if needed, to continue development and using it as CI.

````
 Code maturity issues
Deprecated member is still used (9)
 Compiler issues
Unchecked warning (Errors) (3)
 Data flow issues
Scope of variable is too broad (Errors) (18)
 Declaration redundancy
Method returns the same value (1)
Redundant throws clause (Errors) (11)
Redundant throws declaration (Errors) (11)
 Dependency issues
Class with too many transitive dependents (Errors) (7)
 Encapsulation issues
'public' field (13)
 General
Annotator (2)
Redundant suppression (Errors) (12)
 Imports
Unused import (965)
 Java language level migration aids
Usages of API which isn't available at the configured language level (Errors) (1183)
 Javadoc issues
Declaration has problems in Javadoc references (53)
Unnecessary {@inheritDoc} Javadoc Comment (Errors) (1)
 Method metrics
Method with too many parameters (Errors) (2)
 Naming conventions
Instance method naming convention (1)
Package naming convention (30)
 Performance issues
Method may be 'static' (51)
 Probable bugs
Result of method call ignored (1)
Unused assignment (Errors) (6)
 Serialization issues
Non-serializable class with 'serialVersionUID' (Errors) (1)
````

**Comment by romani** [comment]
master validation is relaunched, to show other in PR that it is not their fault.

**Comment by rnveach** [comment]
Errors may not be valid. An issue has been filed.
https://teamcity-support.jetbrains.com/hc/en-us/requests/823672
https://youtrack.jetbrains.com/issue/TW-48065

> A lot of ... indicates that project was not correctly imported

**Comment by romani** [comment]
From support:
```
Vladislav Rassokhin­ (TeamCity)­
Dec 7, 18:25 MSK ­

Hi,­

Unfortunately I've to­ revert that upgrade ­to IDEA 2016.3 due to­ problem with inspecti­ons. I'm sure you've alr­eady seen that. We wi­ll investigate furthe­r with IDEA team and ­then would again try ­to migrate to IDEA 20­16.3.

Vladislav Rassokhin­
Java developer (Team­City)
JetBrains­
```

**Comment by romani** [comment]
teamcity is upgraded to new IntelijIdea

in recent PR #4108 :
>` [00:38:20] :	 [Step 2/2] Starting up IntelliJ IDEA 2017.1.1 EAP (build IU-171.4011) ...done.`

so we have to fix all that violations asap or suppress .... to unblock PRs .
Next build on master will fail for sure.

**Comment by rnveach** [comment]
Errors don't look valid again.
I re-posted issue to original issue tracker, https://youtrack.jetbrains.com/issue/TW-48065 .

**Comment by romani** [comment]
New request to support: https://teamcity-support.jetbrains.com/hc/en-us/requests/919114

**Comment by romani** [comment]
I disabled TC triggers, I hope it will stop to appear in Github PR validations.

**Comment by romani** [comment]
>
```
Apr 10, 21:10 MSK
Hi Roman,

We've updated teamci­ty.jetbrains.com, now it uses IntelliJ IDEA 2016.2.5
 to run inspections & duplic­ates so problem with too many inspections 
should be solved.

Vladislav Rassokhin
```

**Comment by romani** [comment]
All triggers are enabled in teamcity service

**Comment by Vladlis** [comment]
I am on it

**Comment by rnveach** [comment]
@romani According to front page banner, TeamCity is now passing.
This issue is finished then, right?

**Comment by romani** [comment]
Yes, let's close it, ones it because problem again we could do reopen or create new issue with new status of validation.
